www/include/views/graphs/exportData/ExportCSVServiceData.php

```diff
@@ -58,31 +58,59 @@ function get_error($str)
     get_error('need session id !');
 }
 
-isset($_GET["index"]) ? $index = htmlentities($_GET["index"], ENT_QUOTES, "UTF-8") : $index = null;
-isset($_POST["index"]) ? $index = htmlentities($_POST["index"], ENT_QUOTES, "UTF-8") : $index = $index;
+$index = filter_var(
+    $_GET['index'] ?? $_POST['index'] ?? null,
+    FILTER_VALIDATE_INT
+);
+$period = filter_var(
+    $_GET['period'] ?? $_POST['period'] ?? 'today',
+    FILTER_SANITIZE_STRING
+);
+$start = filter_var(
+    $_GET['start'] ?? null,
+    FILTER_VALIDATE_INT
+);
+$end = filter_var(
+    $_GET['end'] ?? null,
+    FILTER_VALIDATE_INT
+);
+$chartId = filter_var(
+    $_GET['chartId'] ?? null,
+    FILTER_SANITIZE_STRING
+);
 
-if (isset($_GET['chartId'])) {
+if (isset($chartId)) {
     list($hostId, $serviceId) = explode('_', $_GET['chartId']);
```

Comment:
this may throw an error is $chartId = "15_" as list won't be able to set serviceId

Reply:
This can't happen.
```
    if (!isset($hostId) || !isset($serviceId)) {
        die('Resource not found');
    }
```
The request will not be executed since we throw an error is serviceId || hostId is not defined

Reply:
I may change this to seperate both values and then do the check on those

Reply:
Okay I've just checked it should not break since the second entry of the array (list) will be empty

http://sandbox.onlinephpfunctions.com/code/2878496991cc362b91eecf82f25190340b02a8e1